solr/core/src/test/org/apache/solr/cloud/TestPullReplica.java

```diff
@@ -97,6 +92,90 @@ public static void tearDownCluster() {
     TestInjection.reset();
   }
 
+  static void waitForDeletion(String collection) throws InterruptedException, KeeperException {
```

Comment:
nit: did this need to move, or can it stay around line 555 to minimize the diff? All of these methods moved around but it looks like they mostly stayed the same. Feel free to move them in a follow on PR :)